packages/components/src/Table/Row/TableRow.js

```diff
@@ -0,0 +1,174 @@
+	const compKey = `${getRowId(rowDataGetter, element, index)}-${columnKey}`;
+	const classnames = classNames(`td-${columnKey}`, theme['tc-table-row-cell']);
+	const dataClassnames = classNames(
+		'tc-table-row-data',
+		theme['tc-table-row-data'],
+		getRowDataClassName(classNameProvider, element, columnKey),
+	);
+	return (
+		<td key={`td-${compKey}`} className={classnames}>
+			<CellComponent
+				key={compKey}
+				element={element}
+				data={getRowData(rowDataGetter, element, columnKey)}
+				className={dataClassnames}
+				extra={getCellComponentExtraProps(rowRenderer, columnKey)}
+			/>
+		</td>
+	);
+}
+
+function getRowClassName(classNameProvider, element) {
+	if (classNameProvider && classNameProvider.getForRow) {
+		return classNameProvider.getForRow(element);
+	}
+	return '';
+}
+
+/**
+ * This component displays the data of an element in a table.
+ * A row is divided in columns, each column displaying an element data.
+ */
+export default class TableRow extends Component {
+	constructor(props) {
+		super(props);
+		this.handleMouseEnter = this.handleMouseEnter.bind(this);
+		this.handleMouseLeave = this.handleMouseLeave.bind(this);
+		this.updateRowRef = this.updateRowRef.bind(this);
+	}
+
+	componentDidMount() {
+		if (this.rowRef != null) {
+			this.rowRef.addEventListener('mouseenter', this.handleMouseEnter);
+			this.rowRef.addEventListener('mouseleave', this.handleMouseLeave);
+		}
+	}
+
+	shouldComponentUpdate(nextProps) {
+		if (this.props.rowRenderer && this.props.rowRenderer.needRowUpdate) {
+			return this.props.rowRenderer.needRowUpdate(nextProps);
+		}
+		return true;
+	}
+
+	componentWillUnmount() {
+		if (this.rowRef != null) {
+			this.rowRef.removeEventListener('mouseenter', this.handleMouseEnter);
+			this.rowRef.removeEventListener('mouseleave', this.handleMouseLeave);
+		}
+	}
+
+	handleMouseEnter() {
+		if (this.props.onEnterRow) {
+			this.props.onEnterRow(this.props.element);
+		}
+	}
+
+	handleMouseLeave() {
+		if (this.props.onLeaveRow) {
+			this.props.onLeaveRow(this.props.element);
+		}
+	}
+
+	updateRowRef(ref) {
+		this.rowRef = ref;
+	}
+
+	render() {
+		const {
+			element,
+			index,
+			classNameProvider,
+			columnKeys,
+			rowDataGetter,
+			rowRenderer,
+		} = this.props;
+		const rowKey = getRowId(rowDataGetter, element, index);
+		const classnames = classNames(
+			'tc-table-row',
+			theme['tc-table-row'],
+			getRowClassName(classNameProvider, element),
+		);
+		return (
+			<tr key={rowKey} className={classnames} ref={this.updateRowRef} data-id={rowKey}>
+				{columnKeys.map(key =>
+					renderRowData(element, index, key, rowDataGetter, classNameProvider, rowRenderer),
+				)}
+			</tr>
+		);
+	}
+}
+
+TableRow.propTypes = {
+	element: PropTypes.object,
+	index: PropTypes.number,
+	classNameProvider: PropTypes.shape({
+		getForTable: PropTypes.func,
+		getForHeader: PropTypes.func,
+		getForRow: PropTypes.func,
+		getForRowData: PropTypes.func,
+	}),
+	columnKeys: PropTypes.array,
```

Comment:
please add a isRequired or add a default value to TableRow props

Reply:
done